src/Merchants/Content/Merchant/Services/CartValidator.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Shopware\Production\Merchants\Content\Merchant\Services;
+
+use Shopware\Core\Checkout\Cart\Cart;
+use Shopware\Core\Checkout\Cart\CartValidatorInterface;
+use Shopware\Core\Checkout\Cart\Error\ErrorCollection;
+use Shopware\Core\Content\Product\ProductEntity;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Shopware\Production\Merchants\Content\Merchant\Exception\CartContainsMultipleMerchants;
+
+class CartValidator implements CartValidatorInterface
+{
+    /**
+     * @var EntityRepositoryInterface
+     */
+    private $productRepository;
+
+    public function __construct(EntityRepositoryInterface $productRepository)
+    {
+        $this->productRepository = $productRepository;
+    }
+
+    public function validate(Cart $cart, ErrorCollection $errorCollection, SalesChannelContext $salesChannelContext): void
+    {
+        $criteria = new Criteria($cart->getLineItems()->getReferenceIds());
```

Comment:
Maybe you should filter here, since this must not all be products - and more important UUID's, wich will throw